dockerpty/pty.py

```diff
@@ -132,14 +132,14 @@ def start(self, **kwargs):
 
 
         mappings = [
-            (pty_stdout, io.Stream(sys.stdout)),
-            (pty_stderr, io.Stream(sys.stderr)),
+            (pty_stdout, io.Stream(sys.stdout), True),
+            (pty_stderr, io.Stream(sys.stderr), True),
         ]
 
         if self.interactive:
-            mappings.insert(0, (io.Stream(sys.stdin), pty_stdin))
+            mappings.insert(0, (io.Stream(sys.stdin), pty_stdin, False))
```

Comment:
Nice! This is along the lines of what I tried when I dug into this problem, and I could never quite get it to work.